guides/source/action_controller_advanced_topics.md

```diff
@@ -0,0 +1,613 @@
+**DO NOT READ THIS FILE ON GITHUB, GUIDES ARE PUBLISHED ON https://guides.rubyonrails.org.**
+
+Action Controller Advanced Topics
+=================================
+
+In this guide, you will learn about some advanced topics related to controllers. After reading this guide, you will know how to:
+
+* Protect against cross-site request forgery.
+* Use Action Controller's built-in HTTP authentication.
+* Stream data directly to the user's browser.
+* Filter sensitive parameters from the application logs.
+* Handle exceptions that may be raised during request processing.
+* Use the built-in health check endpoint for load balancers and uptime monitors.
+
+--------------------------------------------------------------------------------
+
+Introduction
+------------
+
+This guide covers a number of advanced topics related to controllers in a Rails
+application. Please see the [Action Controller
+Overview](action_controller_overview.html) guide for an introduction to Action
+Controllers.
+
+Authenticity Token and Request Forgery Protection
```

Comment:
I see what you're saying. Here's what I'm thinking:

- add protect_from_forgery to this section
- I read the security guide CSRF section, and it focuses on explaining what CSRF is, which is good. It has much more details. We link to this section of the security guide already so that's good
- Since this is an "advanced" guide, I think it's okay to leave this info here. True it's more of an FYI, the user doesn't need to *do* anything but it's for them to understand what Rails does by default.

Let me know if you feel differently or want to make any other changes!

Reply:
I separated the Action View methods in a subsection. I could remove this altogether also.